sdk/python/kfp/v2/components/experimental/pipeline_param.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Definition of PipelineParam."""
+
+import json
+import re
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+from kfp.v2.components import utils
+from kfp.v2.components.types import type_utils
+
+
+@dataclass
+class ConditionOperator:
+    """Represents a condition expression to be used in dsl.Condition().
+
+    Attributes:
+      operator: The operator of the condition.
+      operand1: The left operand.
+      operand2: The right operand.
+    """
+    operator: str
+    operand1: Union['PipelineParam', type_utils.PARAMETER_TYPES]
+    operand2: Union['PipelineParam', type_utils.PARAMETER_TYPES]
+
+
+class PipelineParam:
```

Comment:
Can we rename this to something else to avoid confusion with the v1 term?

Reply:
Synced offline, will rename it to `PipelineChannel`